packages/gatsby-image/src/index.js

```diff
@@ -206,22 +207,23 @@ function generateBase64Sources(imageVariants) {
 }
 
 function generateNoscriptSource({ srcSet, srcSetWebp, media, sizes }, isWebp) {
-  const src = isWebp ? srcSetWebp : srcSet
-  const mediaAttr = media ? `media="${media}" ` : ``
-  const typeAttr = isWebp ? `type='image/webp' ` : ``
-  const sizesAttr = sizes ? `sizes="${sizes}" ` : ``
+  const props = {
+    media,
+    sizes,
+    srcset: isWebp ? srcSetWebp : srcSet,
+    type: isWebp ? `image/webp` : undefined,
+  }
 
-  return `<source ${typeAttr}${mediaAttr}srcset="${src}" ${sizesAttr}/>`
+  return <source {...props} />
 }
```

Comment:
This change is redundant. Use `generateImageSources()` instead if string generation is not needed anymore(requires min version of React for the package to be raised to `>=16.6`, last I knew that change is delayed until the next major version.